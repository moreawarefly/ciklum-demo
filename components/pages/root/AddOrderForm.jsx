import React, { useState } from 'react';
// import PropTypes from 'prop-types';
import styled from 'styled-components';
import TextInput from '../../atoms/TextInput';
import Dropdown from '../../atoms/Dropdown';
import SubmitInput from '../../atoms/SubmitInput';
import * as formConfig from '../../../config/AddOrderFormConfig';
import useCurrencyPairs from '../../../services/useCurrencyPairs';

const Form = styled.form`

`;

function AddOrderForm() {
  const [pair, setPair] = useState();
  const [side, setSide] = useState();
  const [orderType, setOrderType] = useState();
  const [limit, setLimit] = useState();
  const [quantity, setQuantity] = useState();
  const pairItems = useCurrencyPairs();

  function createChangeHandler(setStateFunction) {
    return (event) => setStateFunction(event.target.value);
  }

  function handleFormSubmit(event) {
    console.log('form submit');
    event.preventDefault();
  }

  return (
    <div>
      {pairItems
        ? (
          <Form onSubmit={handleFormSubmit}>
            <div>
              pair: {pair}
              side: {side}
              orderType: {orderType}
              limit: {limit}
              quantity: {quantity}
            </div>
            <div>
              <Dropdown
                label={formConfig.pairDropdown.label}
                items={pairItems}
                onChangeHandler={createChangeHandler(setPair)}
              />
            </div>
            <div>
              <Dropdown
                label={formConfig.sideDropdown.label}
                items={formConfig.sideDropdown.items}
                onChangeHandler={createChangeHandler(setSide)}
              />
            </div>
            <div>
              <Dropdown
                label={formConfig.orderTypeDropdown.label}
                items={formConfig.orderTypeDropdown.items}
                onChangeHandler={createChangeHandler(setOrderType)}
              />
            </div>
            <div>
              <TextInput
                label={formConfig.limitInput.label}
                onChangeHandler={createChangeHandler(setLimit)}
              />
            </div>
            <div>
              <TextInput
                label={formConfig.quantityInput.label}
                onChangeHandler={createChangeHandler(setQuantity)}
              />
            </div>
            <div>
              <SubmitInput
                label={formConfig.submitInput.label}
              />
            </div>
          </Form>
        )
        : (
          <div>Loading</div>
        )
      }
    </div>
  );
}

// AddOrderForm.defaultProps = {
//   pairItems: [],
// };

// AddOrderForm.propTypes = {
//   pairItems: PropTypes.arrayOf({
//     key: PropTypes.string,
//     value: PropTypes.string,
//     label: PropTypes.string,
//   }),
// };

export default AddOrderForm;
