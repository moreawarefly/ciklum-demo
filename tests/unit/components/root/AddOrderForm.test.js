/**
 * @jest-environment jsdom
*/

import React from 'react';
import renderer from 'react-test-renderer';
import AddOrderForm from '../../../../components/pages/root/AddOrderForm';

global.fetch = require('jest-fetch-mock');

describe('AddOrderForm', () => {
  beforeEach(() => {
    const fakeData = JSON.stringify([{"pair":"btcusd","price_precision":5,"initial_margin":"30.0","minimum_margin":"15.0","maximum_order_size":"2000.0","minimum_order_size":"0.004","expiration":"NA","margin":true}]);
    fetch.resetMocks();
    fetch.mockResponse(fakeData);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<AddOrderForm addOrderHandler={() => {}}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('calls handler with order object', () => {
  //   const handler = spy();
  //   const wrapper = mount(<AddOrderForm addOrderHandler={handler} />);
  //   wrapper.find('[type="submit"]').get(0).click();
  //   expect(handler).to.have.been.called();
  // });
});
