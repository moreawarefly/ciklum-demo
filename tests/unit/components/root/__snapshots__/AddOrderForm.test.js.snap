// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddOrderForm renders correctly 1`] = `
<div>
  Loading
</div>
`;
